client/src/main/java/org/apache/nemo/client/ClientEndpoint.java

```diff
@@ -154,7 +146,7 @@ public final Enum waitUntilJobFinish(final long timeout,
       // The driver endpoint is not connected yet.
       final long currentNano = System.nanoTime();
       final boolean driverIsConnected;
-      if (DEFAULT_DRIVER_WAIT_IN_MILLIS < unit.toMillis(timeout)) {
+      if (timeout > 0 && DEFAULT_DRIVER_WAIT_IN_MILLIS < unit.toMillis(timeout)) {
```

Comment:
Why don't you just call `waitUntilJobFinish(DEFAULT_DRIVER_WAIT_IN_MILLIS, ...)` and remove this `if` statement?

Reply:
🆗

Reply:
Oh here, it actually calls for the max value between the DEFAULT value and timeout, so I'll implement it accordingly.